packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useCallback, SetStateAction, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { transparentize } from 'polished';
+import facepaint from 'facepaint';
+import Portal from '@leafygreen-ui/portal';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx, keyframes } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+// breakpoints for different screen sizes
+const small = '767px'; // mobile screens, from 0px - 767px
+const medium = '768px'; // tablet screens, from 768px - 1024px
+const large = '1025px'; // laptops/desktop screens, from 1025px and above
+
+export const mq = facepaint([
+  `@media only screen and (max-width: ${small})`,
+  `@media only screen and (min-width: ${medium})`,
+  `@media only screen and (min-width: ${large})`,
+]);
+
+const defaultSpacing = 18;
+
+const fadein = keyframes`
+  from {
+    margin-top: 50px;
+    opacity: 0;
+  }
+  to {
+    margin-top: 0;
+    opacity: 1;
+  }
+`;
+
+const mainContainer = css`
+  animation: fade-in 250ms ease-in-out;
```

Comment:
Let's have a consistent timing on our animations.

Reply:
Got it